### ISSUE TITLE
[Feature] Support `source` option for additional config files

### ISSUE THREAD

**Issue by rdvm** [hyprwm/hypridle#33]
In `hyprland.conf` I source other config files so I can have some device-specific configurations while sharing most of my config between systems. I'd like to have different idle behavior on my laptop vs. my desktop, so it'd be great if that were an option for hypridle as well.

```sh
source=~/.config/hypr/hosts/laptop/hypridle.conf
```

```sh
[ERR] Config has errors:
Config error in file /home/rdvm/dotfiles/hyprland/.config/hypr/hypridle.conf at line 1: config option <source> does not exist.
```

Thanks for all your work on _hyprwm_ !

**Comment by Rabcor** [comment]
yeah this would be great, if only just so you can keep all your power related settings in one config file. There are ways around this but none are pretty.

**Comment by BinaryInk** [comment]
Throwing my hat in the ring for this feature. Adding my additional reasons from #90 (duplicate, closed).

---

1.    This is a common behavior for a plethora of Linux applications. Those that don't support it out of the box typically provide a mechanism (like hyprland itself) to include files (e.g., nginx).
2.    This is supported behavior in hyprland.conf. It would stand to reason that one could expect this behavior in hyprland's other optional daemons.

3.    Personally, I like being able to separate out different bits of configuration files into their own parts for easier maintenance when possible. I doubt I'm entirely alone on this.
4.    Again personally, I use Syncthing to keep my laptop and desktop hyprland configs in sync. Having the source keyword allows me to do the following:

``` bash
# This file must contain a variable of $hostname. 
# This is used to load the machine-specific confs at the end of this script.
# This file should not be synced with Syncthing.
source = ~/.config/hypr/hyprland.conf.d/hostname.conf

...

# Load all loose conf files
source = ~/.config/hypr/hyprland.conf.d/*.conf
# Load machine-specific conf files
source = ~/.config/hypr/hyprland.conf.d/$hostname/*.conf
```

---

(I may look into submitting a PR myself if I get around to comparing the code from hyprland and hypridle)

**Comment by soymadip** [comment]
Hey i upvote for this feature too. 

without this, one unified settings file is not possible too. 

i opened an issue in hyprlock too:- https://github.com/hyprwm/hyprpaper/issues/274

**Comment by martabal** [comment]
> Hey i upvote for this feature too. 
> 
> without this, one unified settings file is not possible too. 
> 
> i opened an issue in hyprlock too:- https://github.com/hyprwm/hyprpaper/issues/274

It's supported since v0.1.7

**Comment by soymadip** [comment]
> > Hey i upvote for this feature too.
> > without this, one unified settings file is not possible too.
> > i opened an issue in hyprlock too:- [hyprwm/hyprpaper#274](https://github.com/hyprwm/hyprpaper/issues/274)
> 
> It's supported since v0.1.7

will this be supported in hyprpaper?

**Comment by soymadip** [comment]
> > Edit:-  Looks like it's implemented. when will be new release made?
> 
> Indeed, it has been added in this commit. [hyprwm/hyprpaper@6502c87](https://github.com/hyprwm/hyprpaper/commit/6502c87e9c70cf61f05160c33eaa4f6b2e621baa). Idk when it will be released, probably when Vaxry wants to. If you can't wait for a new release, you can compile it yourself

yeah just did it, for now localy built and put in ~/.local/bin and using it
